### ISSUE TITLE
unsized strings and vectors don't work

### ISSUE THREAD

**Issue by 3b** [Shinmera/binary-structures#3]
```lisp
(define-io-structure a
  (a (vector (unsigned-byte 8))))
```
or 
```lisp
(define-io-structure a
  (a string))
```

error during compilation with 

```
No io-type with name
  *
```

(and code for `string`s looks like it would inherit the "until end of input" behavior from `vector` rather than "until 0-termination byte" implied by docs)

**Comment by Shinmera** [comment]
Yeah the issue here is that the backend assumes it can always know the size of the vector ahead of time, so it can allocate it once and then fill it. For some backends it can estimate the size based on the remaining space, but for non-file streams for instance it wouldn't know how much is left, requiring a much more expensive path.
